rmw_fastrtps_shared_cpp/src/rmw_response.cpp

```diff
@@ -92,7 +92,9 @@ __rmw_send_response(
   assert(info);
 
   eprosima::fastrtps::rtps::WriteParams wparams;
-  memcpy(&wparams.related_sample_identity().writer_guid(), request_header->writer_guid,
+  memcpy(
+    static_cast<void *>(&wparams.related_sample_identity().writer_guid()),
+    request_header->writer_guid,
```

Comment:
@ivanpauno it seems to me that these structures have a copy assignment operator, why not using that instead?

Reply:
Hmm, after a quick sync with @ivanpauno, this is just another case of a `memcpy` from a primitive array into a `struct` (!?). It's interesting the compiler didn't complain before. Why is this being done across `rmw_fastrtps` instead of defining an adequate copy-assignment operator for `GUID_t`? @richiware @MiguelCompany by any chance, do you happen to know/recall?

Reply:
I solved the problem in https://github.com/ros2/rmw_fastrtps/pull/346/commits/5ebac808ea5e89e7bc8e174e267eb79014fec88f, instead of ignoring the warning as before.